<launch>

  <arg name="debug" default="false"/>
  <arg if="$(arg debug)" name="launch_prefix" value="xterm -rv -e gdb -ex run -args"/>
  <arg unless="$(arg debug)" name="launch_prefix" value=""/>

  <node pkg="openni_camera" type="openni_node" name="openni_node_top" output="screen" launch-prefix="$(arg launch_prefix)">
    <!--param name="topic" value="first" /-->
    <!--param name="device_id" value="2@3" --> <!-- this line uses device on usb bus 2 and addres 3 -->
    <!--param name="device_id" value="B00367707227042B"/--> <!-- this line uses device with given serial number -->
    <param name="device_id" value="#1"/> <!-- this line uses first enumerated device -->
    <remap from="camera" to="camera_top"/>
    <remap from="camera_info" to="camera_top_info"/>
    <param name="rgb_frame_id" value="/camera_rgb_optical_frame" />
    <param name="depth_frame_id" value="/camera_depth_optical_frame" />
    <param name="use_indices" value="false" />
    <param name="depth_registration" value="true" />
    <param name="image_mode" value="2" />
    <param name="depth_mode" value="2" />
    <param name="debayering" value="2" />
    <param name="depth_time_offset" value="0" />
    <param name="image_time_offset" value="0" />
  </node>

  <node pkg="tf" type="static_transform_publisher" name="camera_rgb_optical_broadcaster" args="0 0 0 0 0 0 camera_rgb_optical_frame openni_rgb_optical_frame_top 100" />
  <node pkg="tf" type="static_transform_publisher" name="camera_depth_optical_broadcaster" args="0 0 0 0 0 0 camera_depth_optical_frame openni_depth_optical_frame_top 100" />

  <include file="$(find openni_launch)/launch/kinect_frames.launch"/>

</launch>
